### PR TITLE
Support T-Test with aggregated data

### PR DESCRIPTION
**Pull request by hideaki**
# Description
Support T-Test with aggregated data.

# Checklist
Make sure you have performed the following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/test_wrapper.R [hunk 1]
```diff
@@ -537,12 +537,281 @@ glance.chisq_exploratory <- function(x) {
...
+    }
+  } else if (alternative == "greater") {
+    res <- 1-p
+  } else {
+    res <- p
+  }
+  res
+}
+
+t.test.aggregated <- function(N1, N2, X1, X2, s1, s2, conf.level=0.95, mu=0, alternative = "two.sided", paired = FALSE, var.equal = FALSE) {
```
Comment (kei51e) [inline R/test_wrapper.R:652]: Add comment.

File: R/test_wrapper.R [hunk 2]
```diff
@@ -537,12 +537,281 @@ glance.chisq_exploratory <- function(x) {
...
+    stderr=stderr,
+    null.value=null.value,
+    method=method,
+    alternative=alternative
+  )
+  class(res) <- c('ttest_exploratory', 'htest')
+  res
+}
+
+exp_ttest_aggregated <- function(df, category, n, category_mean, category_sd, test_sig_level = 0.05,
```
Comment (kei51e) [inline R/test_wrapper.R:691]: Add comment.

File: R/util.R [hunk 3]
```diff
@@ -2146,6 +2156,20 @@ calculate_cohens_d <- function(var1, var2) {
   cd  <- md/csd # cohen's d
 }
 
+calculate_cohens_d_aggregated <- function(N1, N2, X1, X2, s1, s2) {
```
Comment (kei51e) [inline R/util.R:2159]: Add comment.

File: R/util.R [hunk 4]
```diff
@@ -2125,6 +2125,16 @@ calculate_common_sd <- function(var1, var2) {
   csd <- sqrt(csd) # common sd computation
 }
 
+calculate_common_sd_aggregated <- function(N1, N2, s1, s2) {
```
Comment (kei51e) [inline R/util.R:2128]: Add comment.

File: tests/testthat/test_test_wrapper.R [hunk 5]
```diff
@@ -9,6 +9,134 @@ test_df$list_c <- as.list(seq(20))
...
 test_df[["with space"]] <- seq(20)
 
+test_that("test t.test.aggregated with two.sided alternative (default)", {
+  test_df <- data.frame(
+    cat=factor(rep(c("cat1", "cat2"), 20), levels = c("cat1", "cat2")),
+    val = rep(seq(10), 2)
+  )
+  test_df2 <- test_df %>% group_by(cat) %>% summarize(n=n(), sd=sd(val), mean=mean(val))
+  res0 <- t.test(data=test_df, val~cat)
+  res <- t.test.aggregated(test_df2$n[1],test_df2$n[2],test_df2$mean[1],test_df2$mean[2],test_df2$sd[1],test_df2$sd[2],0.95,0)
```
Comment (kei51e) [inline tests/testthat/test_test_wrapper.R:19]: Add namespace for t.test and t.test.aggregated.

File: R/test_wrapper.R (hunk 1 above)
Comment (hideaki) [reply R/test_wrapper.R:652]: Done.

File: R/test_wrapper.R (hunk 2 above)
Comment (hideaki) [reply R/test_wrapper.R:691]: Done.

File: R/util.R (hunk 4 above)
Comment (hideaki) [reply R/util.R:2128]: Done.

File: R/util.R (hunk 3 above)
Comment (hideaki) [reply R/util.R:2159]: Done.

**Review by kei51e — APPROVED** [review]

**Pull request merged**